src/uct/ib/base/ib_md.h

```diff
@@ -357,13 +379,34 @@ typedef struct uct_ib_md_ops_entry {
     int                         priority;
 } uct_ib_md_ops_entry_t;
 
+#define uct_ib_md_ops_register(_name) \
+    uct_ib_md_ops_##_name##_ctor();
+
+#define UCT_IB_MD_OPS_REG_DECL(_name) \
+    void uct_ib_md_ops_##_name##_ctor(void);
+
+#define UCT_IB_MD_OPS_INIT(_md_ops, _priority, _name) \
+    void uct_ib_md_ops_##_name##_init() { \
+        static uct_ib_md_ops_entry_t *p, entry = { \
+            .name     = UCS_PP_MAKE_STRING(_md_ops), \
+            .ops      = &_md_ops, \
+            .priority = _priority \
+        }; \
+        ucs_list_for_each(p, &uct_ib_md_ops_list, list) { \
+            if (p->priority < _priority) { \
+                ucs_list_insert_before(&p->list, &entry.list); \
+                return; \
+            } \
+        } \
+        ucs_list_add_tail(&uct_ib_md_ops_list, &entry.list); \
```

Comment:
can this be a helper function outside the macro?

Reply:
> can this be a helper function outside the macro?

technically yes, but in this case we will need more code to define uct_ib_md_ops_entry_t entry.
what is wrong with current solution?

Reply:
better put as less pas possible inside of slow-path macros - less generated code, easier to debug

Reply:
> better put as less pas possible inside of slow-path macros - less generated code, easier to debug

split it to
```
UCT_IB_MD_OPS_ENTRY \
   static uct_ib_md_ops_entry_t entry = { \
             .name     = UCS_PP_MAKE_STRING(_md_ops), \
             .ops      = &_md_ops, \
             .priority = _priority \
         };
```
and
```
UCT_IB_MD_OPS \
   body
```
???

Reply:
something like 
```c
uct_ib_md_ops_add(uct_ib_md_ops_entry_t *entry)
{
    ucs_list_for_each(p, &uct_ib_md_ops_list, list) {
        if (p->priority < entry->priority) {
                ucs_list_insert_before(&p->list, &entry->list);
                return;
        }
    }
    ucs_list_add_tail(&uct_ib_md_ops_list, &entry->list);
}
````

Reply:
> something like

ok, done